### PR TITLE
Admin chat

### PR DESCRIPTION
**Pull request by ike709**
Adds a chat channel for admins. Messages can be sent with the `asay` command or the `]` hotkey.

Admins also have a button to filter admin chat, the same as normal players filter Local/OOC. Regular players don't have the button.

### CODE REVIEW

**Comment by ike709** [comment]
image

image

**Comment by ike709** [comment]
I've changed it to show username instead of character name.

File: Content.Client/State/GameScreen.cs [hunk 1]
```diff
@@ -27,7 +29,15 @@ public override void Startup()
...
 
-            _gameChat = new ChatBox();
+            if(_groupController.CanCommand("asay"))
+            {
+                _gameChat = new ChatBox(true);
+            }
+            else
+            {
+                _gameChat = new ChatBox(false);
+            }
```
Comment (Zumorica) [inline Content.Client/State/GameScreen.cs:39]: Why not just do this inside ChatBox's constructor so you don't have to pass a bool to it?

Comment (ike709) [reply Content.Client/State/GameScreen.cs:39]: An excellent point.

**Comment by ike709** [comment]
image

File: Content.Server/Chat/ChatManager.cs [hunk 2]
```diff
@@ -112,6 +114,24 @@ public void SendDeadChat(IPlayerSession player, string message)
...
+                return;
+            }
+            var clients = _playerManager.GetPlayersBy(x => _conGroupController.CanCommand(x, "asay")).Select(p => p.ConnectedClient);;
+
+            var msg = _netManager.CreateNetMessage<MsgChatMessage>();
+
+            msg.Channel = ChatChannel.AdminChat;
+            msg.Message = message;
+            msg.MessageWrap = $"{_localizationManager.GetString("ADMIN")}: {player.SessionId}: {{0}}";
+            msg.SenderEntity = player.AttachedEntityUid.GetValueOrDefault();
```
Comment (PJB3005) [inline Content.Server/Chat/ChatManager.cs:131]: Don't bother sending `SenderEntity` here, it's intended for attaching speech bubbles and such which obviously isn't applicable.

Comment (ike709) [reply Content.Server/Chat/ChatManager.cs]: Done.

**Pull request merged**